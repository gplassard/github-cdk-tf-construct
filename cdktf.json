{
  "language": "typescript",
  "app": "npx ts-node bin/github-repository.ts",
  "projectId": "2924e279-9a89-4264-be09-917cfbaf1f77",
  "sendCrashReports": "false",
  "terraformProviders": [],
  "terraformModules": [],
  "context": {
    "excludeStackIdFromLogicalIds": "true",
    "allowSepCharsInLogicalIds": "true"
  }
}