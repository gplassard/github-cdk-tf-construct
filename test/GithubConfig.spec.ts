import { Testing } from 'cdktf';
import { GithubConfig } from '../src/GithubConfig';

describe('GithubConfig', () => {
  it('should produce the expected', () => {
    const app = Testing.app();
    const stack = new GithubConfig(app, 'github');
    const synthesized = Testing.synth(stack);

    expect(synthesized).toMatchSnapshot();
  });
});
