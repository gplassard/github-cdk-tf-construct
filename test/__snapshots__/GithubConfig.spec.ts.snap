// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GithubConfig should produce the expected 1`] = `
"{
  "provider": {
    "github": [
      {
        "owner": "gplassard"
      }
    ]
  },
  "resource": {
    "github_branch_default": {
      "defaultBranch": {
        "branch": "main",
        "repository": "\${github_repository.repo.id}"
      }
    },
    "github_branch_protection": {
      "mainProtection": {
        "allows_deletions": false,
        "allows_force_pushes": false,
        "enforce_admins": true,
        "pattern": "main",
        "repository_id": "\${github_repository.repo.id}",
        "required_linear_history": true,
        "required_pull_request_reviews": [
          {
            "required_approving_review_count": 0
          }
        ],
        "required_status_checks": [
          {
            "contexts": [
              "build",
              "Validate PR title"
            ],
            "strict": true
          }
        ]
      }
    },
    "github_repository": {
      "repo": {
        "allow_auto_merge": true,
        "allow_merge_commit": false,
        "allow_rebase_merge": false,
        "allow_squash_merge": true,
        "allow_update_branch": true,
        "archive_on_destroy": true,
        "delete_branch_on_merge": true,
        "name": "github-cdk-tf-construct",
        "squash_merge_commit_message": "COMMIT_MESSAGES",
        "squash_merge_commit_title": "COMMIT_OR_PR_TITLE",
        "visibility": "public"
      }
    }
  },
  "terraform": {
    "backend": {
      "s3": {
        "bucket": "terraform-state-eu-west-1-1475",
        "key": "github-cdk-tf-construct/terraform.tfstate",
        "region": "eu-west-1"
      }
    },
    "required_providers": {
      "github": {
        "source": "integrations/github",
        "version": "6.2.3"
      }
    }
  }
}"
`;
